### PR TITLE
UIDATIMP-296: Data Import Field Mapping Profile details: Create/Edit Invoice and Invoice line from EDIFACT Invoice

### PR DESCRIPTION
**Pull request by IhorKoba**
## Purpose

To record the details of field mappings and defaults required to create an Invoice and its Invoice lines from an incoming EDIFACT invoice file

## Approach

- Add required permissions to package.json
- Update  `AcceptedValuesField` component to have a possibility to get options from parsed from string value
- Fix proptypes for React Components in `withReferenceValues` and `DatePickerDecorator`
- Add missed key prop in `MappedHeader`
- Update `RepeatableActionsField` component to have a possibility to pass custom options list
- Fix styles for Checkboxes inside `MappingProfiles` form
- Create required accordions for `InvoiceDetailSection` in edit/create mode
- Add translations for `InvoiceDetailSection`

## Refs

https://issues.folio.org/browse/UIDATIMP-296

## Screenshots

image
image
image
image
image

### CODE REVIEW

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import {
+  Accordion,
+  Row,
+  Col,
+  TextField,
+  Checkbox,
+  useCurrencyOptions,
+} from '@folio/stripes/components';
+
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js:17]: Remove this blank line between folio imports, please.

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import { PAYMENT_METHOD_OPTIONS } from '@folio/stripes-acq-components';
+
+import { AcceptedValuesField } from '../../../../../components';
+
+import { getFieldName } from '../../utils';
+import { TRANSLATION_ID_PREFIX } from '../../constants';
+
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js:24]: Remove this blank line between utils imports, please.

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+  Accordion,
+  Row,
+  Col,
+  Card,
+  IconButton,
+  RepeatableField,
+  TextField,
+  Checkbox,
+} from '@folio/stripes/components';
+
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:19]: Remove the blank line

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+  const renderAdjustment = (field, index) => {
+    return (
+      <Card
+        headerEnd={(
+          <IconButton
+            data-test-repeatable-field-remove-item-button
+            icon="trash"
+            onClick={() => onRemove(index, adjustments, 15, setReferenceTables, 'order')}
+            size="medium"
+            ariaLabel={formatMessage({ id: 'stripes-components.deleteThisItem' })}
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:138]: Move it to a variable, please.

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+          />
+        )}
+        headerStart={(
+          <FormattedMessage
+            id={`${TRANSLATION_ID_PREFIX}.invoice.invoiceAdjustments.adjustments.sectionTitle`}
+            values={{ index: index + 1 }}
+          />
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:145]: And this one

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceInformation.js [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+  const [isLockTotal, setIsLockTotal] = useState(false);
+
+  return (
+    <Accordion
+      id="invoice-information"
+      label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.invoice.invoiceInformation.section`} />}
+    >
+      <Row left="xs">
+        <Col
+          data-test-invoice-date
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceInformation.js:40]: Are you sure you need this data-attribute? If you want to test it with RTL it's better to replace it with `data-testid` attribute.

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  Accordion,
+  Row,
+  Col,
+  Card,
+  IconButton,
+  RepeatableField,
+  TextField,
+  Checkbox,
+} from '@folio/stripes/components';
+
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:19]: Blank line

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    return (
+      <Card
+        headerEnd={(
+          <IconButton
+            data-test-repeatable-field-remove-item-button
+            icon="trash"
+            onClick={() => onRemove(index, lineAdjustments, 41, setReferenceTables, 'order')}
+            size="medium"
+            ariaLabel={formatMessage({ id: 'stripes-components.deleteThisItem' })}
+          />
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:57]: Move to a variable, please

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+        )}
+        headerStart={(
+          <FormattedMessage
+            id={`${TRANSLATION_ID_PREFIX}.invoice.invoiceAdjustments.adjustments.sectionTitle`}
+            values={{ index: index + 1 }}
+          />
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:63]: Move to a variable, please

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+              label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.invoice.invoiceAdjustments.field.amount`} />}
+              name={getSubfieldName(41, 1, index)}
+            />
+          </Col>
+          <Col xs={2}>
+            <Field
+              component={TypeToggle}
+              label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.invoice.invoiceAdjustments.field.type`} />}
+              name={getSubfieldName(41, 2, index)}
+              currency={undefined}
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:86]: Do you really need to pass `undefined` to this prop?

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineFundDistribution.js [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+import { Field } from 'redux-form';
+
+import {
+  Accordion,
+  Row,
+  Col,
+  RepeatableField,
+  TextField,
+} from '@folio/stripes/components';
+
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineFundDistribution.js:13]: Blank line

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineInformation.js [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+} from '@folio/stripes/components';
+
+import {
+  AcceptedValuesField,
+  DatePickerDecorator,
+} from '../../../../../components';
+
+import { getFieldName } from '../../utils';
+import { TRANSLATION_ID_PREFIX } from '../../constants';
+
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineInformation.js:20]: Blank line

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/index.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+export * from './InvoiceInformation';
+export * from './InvoiceAdjustments';
+export * from './VendorInformation';
+export * from './ExtendedInformation';
+export * from './InvoiceLineInformation';
+export * from './InvoiceLineFundDistribution';
+export * from './InvoiceLineAdjustments';
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/index.js:7]: Set exports in alphabetical order, please.

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js (hunk 1 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js:17]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js (hunk 2 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/ExtendedInformation.js:24]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js (hunk 3 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:19]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js (hunk 4 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:138]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js (hunk 5 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:145]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceInformation.js (hunk 6 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceInformation.js:40]: Forgot to remove. Done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js (hunk 7 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:19]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js (hunk 8 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:57]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js (hunk 9 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:63]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js (hunk 10 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:86]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineFundDistribution.js (hunk 11 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineFundDistribution.js:13]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineInformation.js (hunk 12 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineInformation.js:20]: done

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/index.js (hunk 13 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/index.js:7]: done

**Review by mariia-aloshyna — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]

**Review by OleksiiKuzminov — APPROVED** [review]

**Pull request merged**